tests/conftest.py

```diff
@@ -3604,7 +3604,9 @@ def empty_base_data_context_in_cloud_mode(
             cloud_mode=True,
             cloud_config=ge_cloud_config,
         )
-    assert context.list_datasources() == []
+    context._datasources = (
+        {}
+    )  # Basic in-memory mock for DatasourceDict to avoid HTTP calls
```

Comment:
Don't think this is a great practice but we can sub in a proper mock/spy once this implementation is fully functional

Reply:
I think this is fine and is one of the benefits of using a `MutableMapping` (or for that matter using any standard python `collections.abc` interfaces).